### PR TITLE
Add extension setting to control showing cli success messages via inf…

### PR DESCRIPTION
**Pull request by lcampos**
…ormation messages or status bar. @W-4440104@.

### What does this PR do?
Adds setting to control showing successful message as information messages or as status bar messages.
### What issues does this PR fix or reference?
Addresses giving the user control on how certain messages are shown.

### CODE REVIEW

File: packages/salesforcedx-vscode-core/package.nls.json [hunk 1]
```diff
@@ -44,5 +44,7 @@
     "SFDX: Execute Anonymous Apex with Currently Selected Text",
   "feature_previews_title": "Salesforce Feature Previews",
   "force_project_create_text": "SFDX: Create Project",
-  "force_apex_trigger_create_text": "SFDX: Create Apex Trigger"
+  "force_apex_trigger_create_text": "SFDX: Create Apex Trigger",
+  "show_cli_success_msg_description":
+    "Specifies whether the SFDX CLI messages executed via vscode will show as information messages (true) or as status bar messages (false)."
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/package.nls.json]: @ruthemmanuelle this text will be part of the default settings file users see above the newly added setting option (salesforcedx-vscode-core.show-cli-success-msg).

File: packages/salesforcedx-vscode-core/src/constants.ts [hunk 2]
```diff
@@ -13,3 +13,6 @@ export const TERMINAL_INTEGRATED_ENVS = [
   'terminal.integrated.env.linux',
   'terminal.integrated.env.windows'
 ];
+export const SFDX_CORE_CONFIGURATION_NAME = 'salesforcedx-vscode-core';
+export const STATUS_BAR_MSG_TIMEOUT_MS = 5000;
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/src/constants.ts]: I've added a default timeout for making the status bar message disappear.

**Comment by Sweetman** [comment]
LGTM. The syntax for the work item linking is a little bit off. Should be @W-XXXXXX@

File: packages/salesforcedx-vscode-core/package.nls.json (hunk 1 above)
Comment (ruthemmanuelle) [reply packages/salesforcedx-vscode-core/package.nls.json]: How about this?
"Specifies whether status messages for Salesforce CLI commands run using the VS Code command palette will appear as pop-up information messages (true) or as status bar messages (false)."

Comment (lcampos) [reply packages/salesforcedx-vscode-core/package.nls.json]: Cool, I'll add that change before merging.

File: packages/salesforcedx-vscode-core/src/notifications/notificationService.ts [hunk 3]
```diff
@@ -74,15 +76,31 @@ export class NotificationService {
     observable: Observable<number | undefined>,
     cancellationToken?: vscode.CancellationToken
   ) {
+    const sfdxCoreSettings = SfdxCoreSettings.getInstance();
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-core/src/notifications/notificationService.ts]: We usually do it a different way where we export the instance directly in a index.ts file. See https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-core/src/notifications/index.ts#L10

File: packages/salesforcedx-vscode-core/src/sfdxCoreSettings.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+import * as vscode from 'vscode';
+import {
+  SFDX_CORE_CONFIGURATION_NAME,
+  SHOW_CLI_SUCCESS_INFO_MSG
+} from './constants';
+/**
+ * A centralized location for interacting with sfdx-core settings.
+ */
+export class SfdxCoreSettings {
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-core/src/sfdxCoreSettings.ts]: Thanks for doing this. Maybe put this under a new settings folder?

**Pull request merged**